### PR TITLE
doc: releases: Add release notes for logging

### PR DESCRIPTION
**Pull request by nordic-krch**
Add section with logging changes.

### CODE REVIEW

File: doc/releases/release-notes-3.6.rst [hunk 1]
```diff
@@ -870,6 +870,29 @@ Libraries / Subsystems
...
 
+* Logging
+
+  * Added option to remove string literals from the binary when dictionary based logging is used.
+
+  * Optimized the most common logging messages (strings with up to 2 numeric arguments). Optimization
+    is done for code size (significant gain seen on riscv32) and performance.
+
+  * Extended logging frontend API to optionally implement dedicated functions for optimized messages.
+    Optional API is enabled by :kconfig:option:`LOG_FRONTEND_OPT_API`.
```
Comment (ycsin) [inline doc/releases/release-notes-3.6.rst:881]: I think it needs the `CONFIG_` prefix to link properly?

```suggestion
    Optional API is enabled by :kconfig:option:`CONFIG_LOG_FRONTEND_OPT_API`.
```

File: doc/releases/release-notes-3.6.rst [hunk 2]
```diff
@@ -870,6 +870,29 @@ Libraries / Subsystems
...
+
+  * Added support for runtime message filtering for the logging frontend.
+
+  * Add option to have multiple instances of the UART logging backend.
+
+  * Fixed userspace issue for :c:func:`printk` when :kconfig:option:`LOG_PRINTK` is enabled.
```
Comment (ycsin) [inline doc/releases/release-notes-3.6.rst:887]:
```suggestion
  * Fixed userspace issue for :c:func:`printk` when :kconfig:option:`CONFIG_LOG_PRINTK` is enabled.
```

**Review by ycsin — DISMISSED** [review]

File: doc/releases/release-notes-3.6.rst (hunk 2 above)
Comment (nordic-krch) [reply doc/releases/release-notes-3.6.rst]: fixed

File: doc/releases/release-notes-3.6.rst (hunk 1 above)
Comment (nordic-krch) [reply doc/releases/release-notes-3.6.rst]: fixed

**Review by henrikbrixandersen — APPROVED** [review]

**Review by ycsin — APPROVED** [review]

**Pull request merged**